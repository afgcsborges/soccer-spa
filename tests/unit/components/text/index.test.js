import React from 'react'
import Text from 'components/text'
import { mount } from 'enzyme'

describe('Text component tests', () => {
    it('should match snapshot with strong and underlined and min width', () => {
        const wrapper = mount(<Text strong label="SOME_LABEL" underline minWidth="100px" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot if no level passed', () => {
        const wrapper = mount(<Text level={null} label="no level" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot level 1', () => {
        const wrapper = mount(<Text level={1} label="level 1" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot level 2', () => {
        const wrapper = mount(<Text level={2} label="level 2" />)

        expect(wrapper).toMatchSnapshot()
    })

    it('should match snapshot level 3', () => {
        const wrapper = mount(<Text level={3} label="level 3" />)

        expect(wrapper).toMatchSnapshot()
    })
})
