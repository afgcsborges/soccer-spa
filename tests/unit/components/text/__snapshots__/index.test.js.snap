// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component tests should match snapshot if no level passed 1`] = `
<Text
  label="no level"
  level={null}
  strong={false}
  underline={false}
>
  <styled.span
    level={null}
    strong={false}
    title="no level"
    underline={false}
  >
    <span
      className="sc-bczRLJ Yipxl"
      title="no level"
    >
      no level
    </span>
  </styled.span>
</Text>
`;

exports[`Text component tests should match snapshot level 1 1`] = `
<Text
  label="level 1"
  level={1}
  strong={false}
  underline={false}
>
  <styled.span
    level={1}
    strong={false}
    title="level 1"
    underline={false}
  >
    <span
      className="sc-bczRLJ Yipxl"
      title="level 1"
    >
      level 1
    </span>
  </styled.span>
</Text>
`;

exports[`Text component tests should match snapshot level 2 1`] = `
<Text
  label="level 2"
  level={2}
  strong={false}
  underline={false}
>
  <styled.span
    level={2}
    strong={false}
    title="level 2"
    underline={false}
  >
    <span
      className="sc-bczRLJ OVqzp"
      title="level 2"
    >
      level 2
    </span>
  </styled.span>
</Text>
`;

exports[`Text component tests should match snapshot level 3 1`] = `
<Text
  label="level 3"
  level={3}
  strong={false}
  underline={false}
>
  <styled.span
    level={3}
    strong={false}
    title="level 3"
    underline={false}
  >
    <span
      className="sc-bczRLJ jOFPme"
      title="level 3"
    >
      level 3
    </span>
  </styled.span>
</Text>
`;

exports[`Text component tests should match snapshot with strong and underlined and min width 1`] = `
<Text
  label="SOME_LABEL"
  level={1}
  minWidth="100px"
  strong={true}
  underline={true}
>
  <styled.span
    level={1}
    minWidth="100px"
    strong={true}
    title="SOME_LABEL"
    underline={true}
  >
    <span
      className="sc-bczRLJ kOfvim"
      title="SOME_LABEL"
    >
      <u>
        SOME_LABEL
      </u>
    </span>
  </styled.span>
</Text>
`;
